### PR TITLE
Fix/reference param issue

### PR DESCRIPTION
**Pull request by brendancolevalley**
The existing codes reference parameter was not able to be leveraged as cloning with depth 1 prevented access to older version. The code was updated to run a full clone, and to checkout the version denoted by `reference` if the parameter is present. If the parameter is not present existing behavior is maintained. 
An additional workflow was added to aid in testing when in a branch (test-trigger-on-push.yaml). 
I verified that including `reference` parameter did checkout the correct micropython repo version, and omitting `reference` cloned the latest commit, and the workflow completed successfully in both cases.

### CODE REVIEW

File: .github/workflows/test-trigger-on-push.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+name: Test on Push Trigger
+
+on:
+  push:
```
Comment (BrianPugh) [inline .github/workflows/test-trigger-on-push.yaml:4]: Why copy/paste all the code from [test.yaml](https://github.com/BrianPugh/install-micropython/blob/main/.github/workflows/test.yaml)?

We can just remove the
```
branches:
 -main
```
specification and delete `test-trigger-on-push.yaml`

**Review by BrianPugh — COMMENTED** [review]
looks good! thank you! Just the comment on modifying CI.

File: .github/workflows/test-trigger-on-push.yaml (hunk 1 above)
Comment (brendancolevalley) [reply .github/workflows/test-trigger-on-push.yaml:4]: There are two differences with new file, the removal of the main branch requirement, and adding in the reference parameter (albeit commented out right now). Wasn't sure of your preference on these two points. Maybe reference parameter can be added to test.yaml in with a value of `v1.22.1` (latest official release)?

Comment (BrianPugh) [reply .github/workflows/test-trigger-on-push.yaml:4]: what if we do something like:

```
jobs:
  test:
    strategy:
      matrix:
        mpy-reference: ["v1.22.1", ""]
    steps:
        ...
      - name: Build and Install MicroPython Unix Port
        uses: ./
        with:
          reference: ${{mpy-reference}}
```

**Comment by brendancolevalley** [comment]
I am not very familiar with github action configuration files. What you propose does not exactly mimic including or excluding the `reference` parameter, but `if (reference)` should evaluate the same so the paths should get exercised. If you want to take this approach,  I can test and update the PR.

**Comment by BrianPugh** [comment]
I agree it's not *exactly* the same, but it should exercise the code the same way (in both cases, `reference` should evaluate to an empty string). `if (reference)` should evaluate as false when provided with an empty string. Let's go with this route, please 😄 . Of course, if the CI behaves unexpectedly, then we'll go with your solution.

**Comment by brendancolevalley** [comment]
I made the changes to test.yaml so that main branch is not required to activate the workflow on push, and we are now running a strategy matrix on `reference`. The tests completed successfully. 
There was one slight adjustment, since `reference` is in a strategy matrix it needs to be referred to with `reference: ${{ matrix.mpy-reference }}`
I have deleted the file test-trigger-on-push.yaml since its no longer required given these changes.

**Pull request merged**